### ISSUE TITLE
vg construct unsupported variant allele

### ISSUE THREAD

**Issue by yihangs** [vgteam/vg#3701]
I use vg (v1.39.0) construct on a vcf file (VCFv4.2) and hg19 but got the following warning: "warning:[vg::Constructor] Unsupported variant allele "<DUP:TANDEM>"; Skipping variant(s)". 

After some simple debugging, I find that it seems that vg cannot support vcf files with ALTs represented by IDs, such as: 
<img width="1363" alt="Screen Shot 2022-07-13 at 10 59 30 AM" src="https://user-images.githubusercontent.com/98494900/178765779-570847a7-df2a-4da4-a295-6040edfa70ce.png">

But will support vcf files with ALTs represented by sequences, such as:
<img width="1363" alt="Screen Shot 2022-07-13 at 11 00 58 AM" src="https://user-images.githubusercontent.com/98494900/178766133-7d63a8e1-a055-4200-9b1b-907a1a9f7d20.png">

Since the former one is a fairly common vcf format, I wonder if there is some way we can do to make vg constrcut also support that kind of vcf file. 

Thanks!

**Comment by glennhickey** [comment]
Have you tried `vg construct -S -f` ?

**Comment by yihangs** [comment]
I tried `vg construct -S -f`, and find that no unsupported variant allele warning appeared anymore. I checked the output .vg file, and find that although this can help encode some ALTs such as `<DEL>` and `<INV>`, it is still not able to support all, e.g. `<DUP:TANDEM>`. More importantly, it seems that vg cannot support all ALTs that are represented by breakends such as `G]chr13:81121773]`, meaning most complex structural variants and inter-chromosomal translocations are removed. Since I am trying to use vg to map reads from some cancer cell types, inter-chromosomal structural variants are actually very important to me. Is there any way to construct a genome graph that has inter-chromosomal SVs? (vg command line or other genome graph construction softwares).
Thank you!

**Comment by glennhickey** [comment]
Yeah, `vg construct -S` only supports `<DEL>, <INV> and <INS>`.  It would be nice to support the full VCF spec, but I don't think anyone's working on that now.   Most of the development is going towards building graphs directly from assemblies without going through VCF.  And while I hope to see pangenome graphs applied to cancer genomes in the future, as far as I know most of the work on whole-genome graphs at present is still focused on germline variation: 

`minigraph` doesn't, as far as I know, presntly support constructing graphs with any inter-chrmosomal events.
`minigraph-cactus` could in theory add some in (if you do not run cactus-graphmap-split) but this hasn't been well tested
`cactus` (without minigraph) may also work if you have very few samples (and all assemblies are run through RepeatMasker first), 

but `PGGB` is probably your best bet, provided you don't want too many samples in your graph (it also relies on chromosome splitting to help scale but can be run without).  Also, it may be difficult to map reads to the graph with current tools..

**Comment by yihangs** [comment]
Thank you for your reply! So vg construct also does not support `<DUP>` (not `DUP:TANDEM`)?

**Comment by maxineliu** [comment]
Hi contributors,
I face similar questions that vg is not able to merge <DUP> and <BND> into graph.vg after using command `vg construct -S`. I wonder if is there any progress on this problem as I notice this issue is not closed.

Hi @yihangs,
I would like to establish a connection with you as I am very interested in your research if you like. Could you email me if you accept my offer? Touch me by email rliu13@uoguelph.ca

**Comment by faithokamoto** [comment]
Seems Glenn answered the questions (and yihangs yes that would be right, I think) and no I don't think anyone was working on this later. Please open a new Issue with updated vg, more details on inputs and commands etc., if this or a similar problem still occurs.

(we don't have automated stale issue closing.)
